### PR TITLE
Polygon.rotate() & Polygon.rotate_ip() - Remake -

### LINKED ISSUE #122: Add polygon rotate(), rotate_ip() functions

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
Takes over #111. Closes #122
I'm making this a draft as the function works but i still need to add some tests.

### CODE REVIEW

**Comment by itzpr3d4t0r** [comment]
This ready now

**Comment by itzpr3d4t0r** [comment]
Ups no not ready, missing tests

File: src_c/polygon.c [hunk 1]
```diff
@@ -7,6 +7,43 @@
 #include <stddef.h>
 #include <math.h>
 
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+#ifndef M_PI_2
+#define M_PI_2 1.57079632679489661923
```
Comment (novialriptide) [inline src_c/polygon.c:15]: We should rename this to `M_PI_QUO_2` to represent that it's the "quotient" for pi and 2.

File: src_c/polygon.c [hunk 2]
```diff
@@ -449,9 +517,137 @@ pg_polygon_collidepoint(pgPolygonObject *self, PyObject *const *args,
...
+    /*Ensure angle is between 0 and two pi*/
+    angle_rad = fmod(angle_rad, M_TWOPI);
+    if (angle_rad < 0) {
+        angle_rad += M_TWOPI;
+    }
+
+    double v1, v2;
+    /*special-cases rotation by 90, 180 and 270 degrees*/
+    switch ((int)(angle_rad / M_PI_2)) {
+        case 1:
```
Comment (novialriptide) [inline src_c/polygon.c:551]: I actually never thought of this. Good job.

**Review by novialriptide — APPROVED** [review]
From a quick skim, LGTM!

File: src_c/polygon.c (hunk 2 above)
Comment (itzpr3d4t0r) [reply src_c/polygon.c:553]: actually it's not my idea, it's something i saw in the pygame.math module in the vector rotate methods ahah.

Comment (itzpr3d4t0r) [reply src_c/polygon.c:553]: it's a bit more complex tho obviously

File: src_c/polygon.c (hunk 1 above)
Comment (itzpr3d4t0r) [reply src_c/polygon.c]: done

**Pull request merged**